### ISSUE TITLE
exporting mermaid chart directly to png

### ISSUE THREAD

**Issue by nrw16** [rich-iannone/DiagrammeR#207]
I would like to export a chart created with `mermaid()` directly to a simple image file (ex: .png) from within an executed .R script without going through the graphical export menus of RStudio.

This has previously come up in the closed issue:  
https://github.com/rich-iannone/DiagrammeR/issues/174 'export_graph does not work with mermaid'

In that issue, an error was produced while trying to use the `export_graph()` function to create .png output from charts generated by the `mermaid()` and `DiagrammeR()` functions.

Issue https://github.com/rich-iannone/DiagrammeR/issues/174 was closed, but I'm still having problems when I try:  

    library("DiagrammeR")
    mg <- mermaid("graph TB
      A-->B")
    export_graph(mg,'mg.png')
Which results in:  
`Error in export_graph(mg, "mg.png") : The graph object is not valid.`

I've tried this with the CRAN-available version of DiagrammeR in both Windows and Linux (debian/kubuntu) environments, inside and outside RStudio, with `DiagrammeRsvg` and `rsvg` packages installed as recommended by error messages.

The chart (in this case, the htmlwidget object `mg`) renders successfully to the RStudio plot window.  From a console session outside RStudio it opens the system's default internet browser window to display the image.  

Despite the htmlwidget object working correctly in those contexts, attempting to use `export_graph()` on it to directly create an image file results in the error described above.

In the `export_graph()` documentation, there is the following line:  
`graph: a graph object of class ‘dgr_graph’.`

Which explains the error message if htmlwidget objects are not supported by this function.

However, back in the closed issue [#174](https://github.com/rich-iannone/DiagrammeR/issues/174) in the comments, it was stated:
>This is currently a limitation but I will work on making 'export_graph()' produce graphical output files with mermaid and Graphviz graph definitions.

And on the [I/O page](http://rich-iannone.github.io/DiagrammeR/io.html) of the DiagrammeR docs:
> Aside from viewing/exporting, there are several workflows for creating DiagrammeR output such as **through R scripts**, R Markdown, and with Shiny apps.  

So it looks like direct from code export (possibly with the `export_graph()` function) is intended to be supported for `mermaid()` charts, but I can't find a description on that page of how to implement any R script workflows, and the `export_graph()` function still seems incompatible with `mermaid()` charts.

Is there something I am missing, that enables a direct export of `mermaid()` charts to simple image file formats like .png?

Is this something that is still under development?

Thanks!

**Comment by rich-iannone** [comment]
@nrw16 you are right in that mermaid graphs aren't supported by `export_graph()`. It's something that I may add in the future however.

**Comment by oscci** [comment]
Hello. Am enjoying using DiagrammeR and have created a nice flowchart with mermaid. It shows up on the viewer in R Studio, but, although the Export tab appears functional, it doesn't work for .png, .tiff or .jpg. It does work for .html. I'd like to use the mermaid graph in a paper that I am creating in RMarkdown that will end up as a Word document. Is there a way to get the diagram into a format for that? Many thanks.  Dorothy

**Comment by alc28** [comment]
I ran into the same problem as oscci yesterday. My clunky workaround was to grab a screenshot of the rendered chart in the rstudio viewer. What is the best practice at this stage of development?

**Comment by senickel** [comment]
Thanks @oscci! This is helped a lot.

The only issue is that I could not edit the size of the image but I created the following work around which does not lead to the exact height defined (slightly bigger) but good enough for me. (on Win7)

```
library(dplyr)
library(DiagrammeR)
library(DiagrammeRsvg)
library(magrittr)
library(svglite)
library(rsvg)
library(png)


# helping function for extracting from list
get.element.from.list<-function (list, el) {
  var <- lapply(list, function(x) return(x[el])) %>% unlist
  return(var)
}


height<-1000 # in pixel

#minimal graph
g1<-"graph g1 {
      A; B
      A--{B}
      }"

exp_g1<-grViz(g1) %>% 
  export_svg 

# getting width height for the ratio
width_height<-exp_g1 %>% 
  strsplit("<svg") %>% 
  get.element.from.list(.,2) %>% 
  strsplit("\n") %>% 
  get.element.from.list(.,1) %>% 
  gsub("\"","",.) %>% 
  strsplit(.," ") %>% 
  unlist %>% 
  gsub("pt","",.) %>% 
  gsub("height=","",.) %>% 
  gsub("width=","",.) %>% 
  data.frame(r=.) %>% 
  filter(r!="") %>% 
  unlist %>% 
  as.numeric

# part 1
p1<-exp_g1 %>% 
  strsplit("<svg") %>% 
  get.element.from.list(.,1) %>% 
  paste0("<svg")

# part 2
# calculate the new width and height
p2<-paste0(" width=\"",
           round((width_height[1]/width_height[2])*height),
           "pt\" height=\"",height,"pt\"")

# part 3
p3<-exp_g1 %>%
  strsplit("<svg") %>%
  get.element.from.list(.,2) %>%
  strsplit("\n viewBox") %>%
  get.element.from.list(.,2) %>%
  paste0("\n viewBox",.)

# put them together
exp_g2<-paste0(p1,p2,p3)

# rest of oscii's work around
exp_g2 %>% 
  charToRaw %>% 
  rsvg %>% 
  png::writePNG("graph.png")

```

**Comment by SebastienBoutry** [comment]
Thanks for the script
